changelog.d/20230715_173829_shane.obrien_TypeName.md

```diff
@@ -0,0 +1,40 @@
+<!--
+A new scriv changelog fragment.
+
+Uncomment the section that is right (remove the HTML comment wrapper).
+-->
+
+<!--
+### Removed
+
+- A bullet item for the Removed category.
+
+-->
+### Added
+
+- `TypeName` record, which gives a richer representation of the components of a PostgreSQL type name (name, schema, modifiers, scalar/array).
+
+### Changed
+
+- `TypeInformation`'s `typeName` parameter from `String` to `TypeName`.
+- `DBEnum`'s `enumTypeName` method from `String` to `QualifiedName`.
+- `DBComposite`'s `compositeTypeName` method from `String` to `QualifiedName`.
```

Comment:
Assuming this is true:

```suggestion
- `TypeInformation`'s `typeName` parameter from `String` to `TypeName`.
- `DBEnum`'s `enumTypeName` method from `String` to `QualifiedName`.
- `DBComposite`'s `compositeTypeName` method from `String` to `QualifiedName`.
- Both `TypeName` and `QualifiedName` have `IsString` instances which have the same behavior as the original `String` types they now replace.
```

Reply:
Eh, kind of. If your string represents an unqualified name with no uppercase or special characters (or modifiers or array nonsense in the case of `TypeName`), then yes. Otherwise, no. Someone could previously have set `typeName` to something like `"mySchema"."myType"(4)[][]` — we make no attempt to parse that into a `typeName`.